### ISSUE TITLE
require-returns-check: False positive with type definition after return

### ISSUE THREAD

**Issue by ilyub** [gajus/eslint-plugin-jsdoc#935]
(This is different from the bug I submitted earlier today for the same rule)

## Expected behavior

Should pass without errors

## Actual behavior

Getting error:
```
JSDoc @returns declaration present but return expression not available
```

## ESLint Config

```js
rules: { "jsdoc/require-returns-check": "warn" }
```

## ESLint sample

```ts
/**
 * Description.
 *
 * @returns Result.
 */
export function f(): string {
  return "";

  type A = string;
}
```

Same problem with interface:
```ts
/**
 * Description.
 *
 * @returns Result.
 */
export function f(): string {
  return "";

  interface I {}
}
```

## Environment

- Node version: 18.12.1
- ESLint version 8.28.0
- `eslint-plugin-jsdoc` version: 39.6.3
